### ISSUE TITLE
Feature request: support for mouse events

### ISSUE THREAD

**Issue by yardnsm** [wix-incubator/unidriver#15]
I'm currently using UniDriver while developing [a new component](https://github.com/wix/wix-style-react/pull/2626/files#diff-4aa809860e3481398d6011ba98b3ed73), where I need to trigger mouse events. Currently I'm using `Simulate` to trigger these, but of course it defeats the whole purpose of UniDriver.

It'll be great to have support for mouse events, such as `mouseDown`, `moueUp`, `mouseEnter`, `mouseLeave`, etc. etc.

Awesome work BTW 😄

**Comment by GabiGrin** [comment]
anything that makes sense telling a user is a great candidate to the API
I think those fit.
We actually had a few places that did that locally in the driver (using the getNative escape hatch)

@Tokyros what did u end up doing?

**Comment by GabiGrin** [comment]
And many thanks for the compliment. Your request will surely make it even better 🌟

**Comment by Tokyros** [comment]
> anything that makes sense telling a user is a great candidate to the API
> I think those fit.
> We actually had a few places that did that locally in the driver (using the getNative escape hatch)
> 
> @Tokyros what did u end up doing?

@GabiGrin 
I implemented functions based on the current .click() method.
e.g.

mouse over:

```
const mouseOver = async (uniDriver: UniDriver) => {
	const el = await uniDriver.getNative();
	if (document.body.contains(el)) {
		const event: MouseEvent  = document.createEvent('MouseEvents');
		event.initEvent('mouseover', true, false);
		el.dispatchEvent(event);
	} else {
		Simulate.mouseEnter(el);
	}
};
```

**Comment by Tokyros** [comment]
> @Tokyros using `dispatchEvent ` will also trigger event bubbling, right? `Simulate` requires you to target the specific node with the events, which can be a pain sometimes.

It should since that's what the second parameter to event.initEvent() does. 
however I did not test such scenario...

**Comment by GabiGrin** [comment]
@yardnsm  we only need `Simulate` if the element is not attached to the dom properly (from our tests it didn't work otherwise, maybe it changed with Reac 16)
But the ideal scenario is to use normal DOM API

**Comment by GabiGrin** [comment]
fixed in #31
